server/src/main/java/io/crate/expression/predicate/NotPredicate.java

```diff
@@ -116,7 +117,7 @@ boolean useNotQuery() {
         }
 
         boolean useFieldExistQuery() {
-            return removeNullValues && !useNotQuery && !enforceThreeValuedLogic;
```

Comment:
This makes sense, because when `removeNullValues` is set we are using a not-query anyway.